tests/test_json_to_parquet.py

```diff
@@ -0,0 +1,531 @@
+                    },
+                    "Grouping": {
+                        "TableGroupingPolicy":"CombineCompatibleSchemas"
+                    }
+                })
+        )
+        glue_client.start_crawler(Name=crawler_name)
+        response = {"Crawler": {}}
+        for i in range(60): # wait up to 10 minutes for crawler to finish
+            # This should take about 5 minutes
+            response = glue_client.get_crawler(Name=crawler_name)
+            if (
+                    "LastCrawl" in response["Crawler"]
+                    and "Status" in response["Crawler"]["LastCrawl"]
+                    and response["Crawler"]["LastCrawl"]["Status"] == "SUCCEEDED"
+               ):
+                break
+            else:
+                time.sleep(10)
+        yield glue_crawler
+        glue_client.delete_crawler(Name=crawler_name)
+
+    @pytest.fixture()
+    def json_s3_objects(self, datadir, artifact_bucket, namespace):
+        s3_client = boto3.client("s3")
+        dataset_prefix = os.path.join(namespace, "tests")
+        object_keys = []
+        for dirpath, _, filenames in os.walk(datadir):
+            if len(filenames) > 0:
+                object_prefix = os.path.join(
+                        dataset_prefix,
+                        os.path.join(*dirpath.split("/")[-6:]))
+                for filename in filenames:
+                    object_key = os.path.join(object_prefix, filename)
+                    s3_client.upload_file(
+                            Filename=os.path.join(dirpath, filename),
+                            Bucket=artifact_bucket,
+                            Key=object_key)
+                    object_keys.append(object_key)
+        return object_keys
+
+    @pytest.fixture(scope="class")
+    def glue_context(self):
+        glue_context = GlueContext(SparkSession.builder.getOrCreate())
+        return glue_context
+
+    def test_upload_test_data(self, glue_test_data):
+        """
+        Upload test data to S3.
+        """
+        pass
+
+    def test_run_glue_crawler(self, glue_crawler):
+        """
+        Create and run Glue crawler.
+        """
+        pass
+
+    def test_get_table(self, glue_database_name, glue_flat_table_name,
+                       glue_nested_table_name, glue_context):
+        flat_table = json_to_parquet.get_table(
+                table_name=glue_flat_table_name,
+                database_name=glue_database_name,
+                glue_context=glue_context
+        )
+        assert flat_table.count() == 1
+        assert len(flat_table.schema().fields) == 2
+        nested_table = json_to_parquet.get_table(
+                table_name=glue_nested_table_name,
+                database_name=glue_database_name,
+                glue_context=glue_context
+        )
+        assert nested_table.count() == 1
+        assert len(nested_table.schema().fields) == 4
+
+    def test_has_nested_fields(self, glue_database_name, glue_flat_table_name,
```

Comment:
Could use something like `@pytest.mark.parametrize` to separate out these
two tests and use the ids parameter to title what each test is testing so it will show up when you run pytest: https://docs.pytest.org/en/6.2.x/parametrize.html

Reply:
Yes, that works. I changed it to use `@pytest.mark.parametrize`.